zcash_client_sqlite/src/lib.rs

```diff
@@ -1180,7 +1202,28 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                 )
             }
 
+            // A flag used to determine whether it is necessary to query for transactions that
+            // provided transparent inputs to this transaction, in order to be able to correctly
+            // recover transparent transaction history.
+            #[cfg(feature = "transparent-inputs")]
+            let mut tx_has_wallet_outputs = false;
+
+            #[cfg(feature = "transparent-inputs")]
+            let detectable_via_scanning = {
+                #[allow(unused_mut)]
+                let mut detectable_via_scanning = d_tx.tx().sapling_bundle().is_some();
+                #[cfg(feature = "orchard")] {
+                    detectable_via_scanning |= d_tx.tx().orchard_bundle().is_some();
+                }
+
+                detectable_via_scanning
+            };
```

Comment:
```suggestion
            #[cfg(feature = "transparent-inputs")]
            let detectable_via_scanning = d_tx.tx().sapling_bundle().is_some();
            #[cfg(all(feature = "transparent-inputs", feature = "orchard"))] {
            let detectable_via_scanning = detectable_via_scanning | d_tx.tx().orchard_bundle().is_some();
```
This is slightly more elegant I think. Non-blocking.